common/Fabric-Install-Utilities.psm1

```diff
@@ -84,11 +84,27 @@ function New-AppRoot($appDirectory, $iisUser){
 }
 
 function New-AppPool($appName, $userName, $credential){
-    Set-Location IIS:\AppPools
+    Set-Location "IIS:\AppPools"
     if(!(Test-Path $appName -PathType Container))
     {
         Write-Console "AppPool $appName does not exist...creating."
-        $appPool = New-WebAppPool $appName
+        $successful = $FALSE
+        $failCount = 0
+        while (!$successful) {
+            try {
+                $appPool = New-WebAppPool $appName -ErrorAction Stop
+                $successful = $TRUE
+            }
+            catch {
+                $failCount = $failCount + 1
+                if ($failCount -gt 3) {
+                    throw
+                }
+
+                Write-Host "Retry #$failCount"
```

Comment:
Can we make this message more specific, like "failed to create appPool $appName $failCount times , retrying..."

Reply:
Thanks for catching that, I meant to fix that and forgot.